en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md

```diff
@@ -0,0 +1,296 @@
+             <p>StarWarsAPI</p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Context</p>
+          </td>
+          <td>
+             <p><code>/swapi</code></p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Version</p>
+          </td>
+          <td>
+             <p>1.0.0</p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Endpoint</p>
+          </td>
+          <td>
+             <a href="http://localhost:8080/graphql" target="_blank">http://localhost:8080/graphql</a>
+          </td>
+          </tr>
+          </table>
+
+       [![Add GraphQL API details]({{base_path}}/assets/img/learn/create-graphql-api-details.png){: style="width:75%"}](../../assets/img/learn/create-graphql-api-details.png)
+
+5. Click `Create` button.
+
+!!! note
+
+    Server implementation relevant to the above API can be obtained from [here]({{base_path}}/tutorials/create-and-publish-a-graphql-api/#step-1-start-the-graphql-backend-server)
+
+
+### Step 2 - Deploy and publish the API
+
+
+1. Navigate to **Deploy** and click **Deployments** to retrieve the API deployment page.
+2. Click **Deploy** to deploy the API in Choreo Connect.
+
+       [![Deploy GraphQL API]({{base_path}}/assets/img/learn/deploy-graphql-api.png)]({{base_path}}/assets/img/learn/deploy-graphql-api.png)
+
+       The Deployment page appears.
+
+       [![GraphQL API Deployment page]({{base_path}}/assets/img/learn/graphql-api-revision-1.png)]({{base_path}}/assets/img/learn/graphql-api-revision-1.png)
+
+3. Navigate to **Publish** and click **Lifecycle**.
+
+       The API lifecycle page appears.
+
+4. Click **Publish** to publish the API to the API Developer Portal.
+
+       [![Publish GraphQL API]({{base_path}}/assets/img/learn/publish-graphql-api.png)]({{base_path}}/assets/img/learn/publish-graphql-api.png)
+
+### Step 3 - Generate an Access Token to invoke the API
+
+1. Click **View in Dev Portal** at the top right corner to open Developer Portal in another browser tab.
+
+2. Open **Applications** from the top menu and select **DefaultApplication** from the list.
+
+3. Open the **Subscriptions** tab and subscribe your API to the **DefaultApplication**. 
+
+4. Open **APIs** from the top menu and select your API.
+
+5. Open the **Try Out** tab and click **GET TEST KEY**. This will include the access token in the cURL command you generate in the section below.
+
+### Step 4 - Invoke the API by providing a valid GraphQL query
+
+1. Since GraphQL APIs require a GraphQL query to provide results, you need to have a valid query. A sample query is provided in below section. In API Manager publisher portal, GraphiQL is used for build queries. You can use explorer option available there to build queries. More information regarding valid GraphQL query generation includes in [here]({{base_path}}/tutorials/create-and-publish-a-graphql-api/#step-51-optionally-try-out-a-query-operation).
```

Comment:
```suggestion
1. Since GraphQL APIs require a GraphQL query to provide results, you need to have a valid query. A sample query is provided in the below section. In API Manager Publisher Portal, GraphQL is used for build queries. You can use the explorer option available there to build queries. For more information regarding valid GraphQL query generation, see [here]({{base_path}}/tutorials/create-and-publish-a-graphql-api/#step-51-optionally-try-out-a-query-operation).
```